autowarefoundation/autoware
# Mxnet trafficlight detection

## Status
**PRODUCTION / DEVELOPMENT**

## Description
- Changed to logic to use filtered_signal_data instead of vectormap server which reduces the execution time to ~300us. 

 This was needed because computing lane to signal light connection is too slow (it slows down the 
 overall traffic light detection to 1Hz) in this function: 
 https://github.com/CPFL/Autoware/blob/feature/pacmod_apex/ros/src/data/packages/vector_map_server/nodes/vector_map_server/vector_map_server.cpp#L357-L363 we go around the problem by providing a 
 list of lights that are on our waypoint. Note: This is a semi-hack that will only work for our demo. If you 
 record different waypoints on different lanes, you need to recreate filteredsignaldata.csv.

 To make the proper fix we would need to use a different data structure instead of 
 https://github.com/CPFL/Autoware/blob/feature/pacmod_apex/ros/src/data/packages/vector_map/include/vector_map/vector_map.h#L362 for faster querying.

- Integrates the `feat_proj` and `region_tlr_mxnet` with runtime manager

## Related PRs

## Todos

## Steps to Test or Reproduce
Play a bag file with image and tf topics or run the follwing in the car.
```
 rosrun runtime_manager runtime_manager_dialog.py 
```
Load `vector_map` 'point_map'.                      
In `computing` tab  under `road_wizard`:
 - select `app` in `feat_proj` and click `use_filtered_signal_map` and load the filtered signal map.
 - select `feat_proj`
 - select `region_tlr_mxnet`

In Rviz slelect image_type with topic `/tlr_superimpose_image` which should show the projected traffic light with the predicted signal
